The purpose of this exercise is to determine how much of each tablet (SuperVit Tablet & NewHealth Tablet) to purchase each day to meet the minimum requirements of nutrition while spending the least amount of money. We will use the glpk and pulp packages to build a model which will give us the most optimal results while taking constraints and rules into consideration.

Please, go to your _environment_ in Anacoda Navigator to install **glpk** and **pulp**  before runing the codes below.
Then, call the library:

In [ ]:
# !pip show  pulp
# !pip install pulp

In [ ]:
import pulp as pp

## 1. **Initialize the MODEL**:

#Define the name & Problem Type.

In [ ]:
model = pp.LpProblem(name='vitamin-problem', # name
                     sense=pp.LpMinimize) # problem type

## 2. **Declare the VARIABLES**:


The supplement model consists of the following decision variables:

In [ ]:
# how much SuperVit?
supervit = pp.LpVariable(name="supervit",  # define the name
                    lowBound=0,  # ensure non-negativity
                    cat='Integer') #Integer value since we are working with quantity.

# how much NewHealth?
newhealth = pp.LpVariable(name="newhealth", # define the name
                 lowBound=0, # ensure non-negativity
                 cat='Integer') #Integer value since we are working with quantity.

## 3. **Create function to OPTIMIZE**:

This function consists of the linear combination of the variables and their _given coefficients_:

In [ ]:
supervitCoeff=0.2 #supervit coefficient
newhealthCoeff=0.3 #newhealth coefficient
obj_func = supervitCoeff*supervit + newhealthCoeff*newhealth #create the function that will calculate the problem's objective solution.

## 4. **Represent the constraints**:

These are the rules the model (set of variables) must obey:

In [ ]:
C1= pp.LpConstraint(name='Vitamin C',   # define the name
                    e= 20 * supervit + 30 * newhealth, #rule
                    rhs=60, #Constraint
                    sense=pp.LpConstraintGE) # >= 'sense'
                    #Same logic for the entire constraints listed below.
C2= pp.LpConstraint(name='Calcium',
                    e= 500 * supervit + 250 * newhealth,
                    rhs=1000,
                    sense=pp.LpConstraintGE)
C3= pp.LpConstraint(name='Iron',
                    e= 9 * supervit + 2 * newhealth,
                    rhs=18,
                    sense=pp.LpConstraintGE)
C4= pp.LpConstraint(name='Niacin',
                    e= 2 * supervit + 10 * newhealth,
                    rhs=20,
                    sense=pp.LpConstraintGE)
C5= pp.LpConstraint(name='Magnesium',
                    e= 60 * supervit + 90 * newhealth,
                    rhs=360,
                    sense=pp.LpConstraintGE)

## 5. **Build MODEL**:

add up the objective functions and all constraints.

In [ ]:
model = pp.LpProblem(name='vitamin-problem', sense=pp.LpMinimize)

model += obj_func
model += C1
model += C2
model += C3
model += C4
model += C5

## 6. **Solve the MODEL**:



In [ ]:
# model.solve()
model.solve(pp.PULP_CBC_CMD(msg=0)) #Execute to solve the model problem.

1

## 7. **Basic Report**


In [ ]:
"Model Status",pp.LpStatus[model.status] #Provide our model's status.

('Model Status', 'Optimal')

In [ ]:
#Results
import pandas as pd
pd.DataFrame.from_dict(Optimal_Decision,orient='index',columns=['info']).map('{:,.2f}'.format) #Create output table with results.

,info
Optimal Solution to minimize spending,1.20
newhealth,2.00
supervit,3.00


Results Summary:

Based on our results, our optimal minimum daily cost will be at $1.20 everyday. We wiill need to purchase 2 newhealth and 3 supervit capsules everyday to meet the minimum nutrition requirements.